cirq-google/cirq_google/engine/stream_manager_test.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import pytest
+import google.api_core.exceptions as google_exceptions
+
+from cirq_google.engine.stream_manager import ResponseDemux
+from cirq_google.cloud import quantum
+
+RESPONSE0 = quantum.QuantumRunStreamResponse(
+    message_id='0', result=quantum.QuantumResult(parent='projects/proj/programs/prog/jobs/job0')
+)
+RESPONSE1 = quantum.QuantumRunStreamResponse(
+    message_id='1', result=quantum.QuantumResult(parent='projects/proj/programs/prog/jobs/job1')
+)
+RESPONSE1_WITH_DIFFERENT_RESULT = quantum.QuantumRunStreamResponse(
+    message_id='1', result=quantum.QuantumResult(parent='projects/proj/programs/prog/jobs/job2')
+)
+
+
+class TestResponseDemux:
+    @pytest.fixture
+    def demux(self) -> ResponseDemux:
+        return ResponseDemux()
+
+    @pytest.mark.asyncio
+    async def test_one_subscribe_one_publish(self, demux):
```

Comment:
Can you add a test that demos the behavior of two subscriptions to the same message id?